pkg/compat/client.go

```diff
@@ -159,9 +176,9 @@ func (c Client) Delete(ctx context.Context, in runtime.Object, opt ...k8sclient.
 // The resource will be converted to a compatible version as needed.
 func (c Client) Update(ctx context.Context, in runtime.Object) error {
 	obj := c.downConvert(in)
+	err := scheme.Scheme.Convert(in, obj, ctx)
+	if err != nil {
+		return err
+	}
 	return c.Client.Update(ctx, obj)
 }
-
-func (c Client) Status() k8sclient.StatusWriter {
```

Comment:
I thought this was needed to implement the `Client` interface.  No?

Reply:
It is inheriting that from the client inside the structure.